library/platform_util.c

```diff
@@ -165,6 +171,7 @@ void *mbedtls_platform_memset( void *ptr, int value, size_t num )
             return ptr;
         }
     }
+    mbedtls_platform_fault();
```

Comment:
Can we define mbedtls_platform_fault() {return NULL} ?
In this case we can remove return from our function in real world . Or just "return mbedtls_platform_fault ()" ?

Reply:
I'm not sure what would the improvement be. This function is only for reporting a failure, and this way we would give the users a possibility to change the return value, which is not what we want to do.

Reply:
I am worried because     mbedtls_platform_fault(); may be skipped and error value will be returned. Not sure there is a reason for my concern

Reply:
But if we move it to a separate function, then inside it this call might be skipped and only the return will be called. It's just moving the place around.

Reply:
I mean if  mbedtls_platform_fault() is defined by us it will return NULL, if by user , it will halt the system

Reply:
I didn't mean to add other function that return NULL

Reply:
The problem is that mbed TLS requires this function to return NULL if allocation doesn't succeed, and delegating it to the application is violating the rules of encapsulation. We can do it if you want, but it seems to me that it's a workaround rather than a solution. Is there a situation when anything else than a NULL would be returned?

Reply:
The function return pointer to buffer if there is no FI problems

Reply:
Yes, but what I mean is - we're returning either a pointer to buffer, or NULL in case of a FI detected. Both of these things are determined inside our library, inside this function - so why should we delegate the decision to the application? It has no new information, it **shouldn't** return anything else than we expect - so this doesn't make sense from the design point of view.

Reply:
Yeah I understand that it is not a "good design" proposal, but I am worried that halt function can be skipped and NULL will be returned. let's give it to Alexander for review and think after some appropriate solution if Alexander will say that there is a problem